airflow/models/taskinstance.py

```diff
@@ -1132,7 +1133,8 @@ def signal_handler(signum, frame):  # pylint: disable=unused-argument
         self._run_execute_callback(context, task)
 
         # Execute the task
-        result = self._execute_task(context, task_copy)
+        with set_current_context(context):
```

Comment:
I am curious what's the contract for `get_current_context` here. For any function/method that takes context argument during task execution, should they expect the passed in context to be the same as what gets returned from `get_current_context`?

Asking this because post_execute callback is not wrapped in the same context that _execute_task is in.

Reply:
Good catch! I missed it although @mik-laj mentioned it here https://github.com/apache/airflow/pull/8651#discussion_r432954063

Reply:
Hm, I think the `get_current_context` is only for "functional dags" (using `@task` to decorate functions) and there's no support for pre a post executes. So in my opinion only `execute` should be wrapped in this context. @casassg

Reply:
Sounds good, just wanted to make sure this expected behavior. Probably worth documenting since the method name itself doesn't convey the message that it should only be used within functional dags.

Reply:
Yeah we should add a note if this is strictly for functional DAG

Reply:
Since it is used only functional DAG, why are we making changes to task instanace?

Reply:
If we can move the logic into functional dag module, that would be definitely more explicit and easier to maintain going forward.

Reply:
Agree. functional dag module is better for that.

Reply:
> Since it is used only functional DAG, why are we making changes to task instanace?

Not sure about this question. We make changes in ti because that's where the logic persists.

Reply:
Yup agree to that. Also, we are modifying task instance due to this being the best module to set the current context (as it's the wrapper for any operator execution.

Regarding wrapping pre_execute/post_execute, we could discuss this but  I think it should be fine for now to keep it at execute (and make sure we write docs about this)

Reply:
@mik-laj @potiuk WDYT?

Reply:
Fine for me